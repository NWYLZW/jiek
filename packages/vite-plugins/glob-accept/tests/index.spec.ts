import fs from 'node:fs'
import path from 'node:path'

import type { ViteDevServer } from 'vite'
import { createServer } from 'vite'
import { afterAll, beforeAll, describe, expect, test } from 'vitest'

import { globAccept, transformImportGlobAccept } from '#~'

const paths = {
  simpleRoot: path.resolve(__dirname, 'fixtures/simple')
}

describe('transformImportGlobAccept', () => {
  test('transformImportGlobAccept', async () => {
    const importer = path.resolve(paths.simpleRoot, 'src/index.ts')
    const simpleCode = fs.readFileSync(importer, 'utf-8')
    const { files } = await transformImportGlobAccept(
      simpleCode,
      importer,
      paths.simpleRoot,
      id => id
    ) ?? {}
    expect(files).toStrictEqual([
      path.resolve(paths.simpleRoot, 'src/a.ts'),
      path.resolve(paths.simpleRoot, 'src/b.ts')
    ])
  })
})

describe('glob-accept', () => {
  let server: ViteDevServer = null as unknown as ViteDevServer
  beforeAll(async () => {
    server = await createServer({
      root: paths.simpleRoot,
      plugins: [globAccept()]
    })
    await server.listen(15487)
  })
  afterAll(async () => {
    await server.close()
  })
  test('create server', async () => {
    const r = await fetch('http://localhost:15487/src/index.ts', { headers: { 'Accept': 'text/plain' } })
    expect(r.status).toBe(200)
    expect(await r.text()).matchSnapshot()
  })
})
