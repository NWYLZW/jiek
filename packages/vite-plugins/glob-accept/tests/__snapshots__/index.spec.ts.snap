// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`glob-accept > create server 1`] = `
"import { createHotContext as __vite__createHotContext } from "/@vite/client";import.meta.hot = __vite__createHotContext("/src/index.ts");import * as __vite_glob_0_0 from "/src/a.ts";import * as __vite_glob_0_1 from "/src/b.ts";/* #__PURE__ */ Object.assign({"./a.ts": __vite_glob_0_0,"./b.ts": __vite_glob_0_1});
if (import.meta.hot) {
  import.meta.hot.accept(["/@fs/Users/yijie/codes/projects/jiek/packages/vite-plugins/glob-accept/tests/fixtures/simple/src/a.ts", "/@fs/Users/yijie/codes/projects/jiek/packages/vite-plugins/glob-accept/tests/fixtures/simple/src/b.ts"], (...args) => console.log.call(void 0, ...args, ["/@fs//Users/yijie/codes/projects/jiek/packages/vite-plugins/glob-accept/tests/fixtures/simple/src/a.ts", "/@fs//Users/yijie/codes/projects/jiek/packages/vite-plugins/glob-accept/tests/fixtures/simple/src/b.ts"]));
}

//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJtYXBwaW5ncyI6IkFBRUEsWUFBWSxLQUFLLFVBQVUsRUFBRSxPQUFPLEtBQUssQ0FBQztBQUMxQyxJQUFJLFlBQVksS0FBSztBQUNuQixjQUFZLElBQUksT0FBTywwZEFBdUI7QUFDaEQiLCJuYW1lcyI6W10sImlnbm9yZUxpc3QiOltdLCJzb3VyY2VzIjpbImluZGV4LnRzIl0sInNvdXJjZXNDb250ZW50IjpbIi8vLyA8cmVmZXJlbmNlIHR5cGVzPVwidml0ZS9jbGllbnRcIiAvPlxuXG5pbXBvcnQubWV0YS5nbG9iKCcuLyoudHMnLCB7IGVhZ2VyOiB0cnVlIH0pXG5pZiAoaW1wb3J0Lm1ldGEuaG90KSB7XG4gIGltcG9ydC5tZXRhLmhvdC5hY2NlcHQoWycuLyoudHMnXSwgY29uc29sZS5sb2cpXG59XG4iXSwiZmlsZSI6Ii9Vc2Vycy95aWppZS9jb2Rlcy9wcm9qZWN0cy9qaWVrL3BhY2thZ2VzL3ZpdGUtcGx1Z2lucy9nbG9iLWFjY2VwdC90ZXN0cy9maXh0dXJlcy9zaW1wbGUvc3JjL2luZGV4LnRzIn0="
`;
